src/Components/List/List.jsx

```diff
@@ -38,18 +68,58 @@ const purchaseItem = (item, token) => {
         lastPurchased: new Date(),
         oldPurchased: item.lastPurchased,
         numberOfPurchases: numberOfPurchases,
-        calculatedEstimate: calculateEstimate(
-          item.frequency,
-          lastInterval,
-          numberOfPurchases,
-        ),
+        calculatedEstimate: calculatedEstimate,
       },
     });
 };
 
+//Color code based on days until next purchase or inactivity with corresponding aria label
+const colorCode = (item) => {
+  const daysSincePurchased = dayjs().diff(
+    getLastPurchaseDate(item, item.lastPurchased),
+    'day',
+  );
+  const estimatedCountdown = item.calculatedEstimate - daysSincePurchased;
+  const elapsedTime = dayjs().isSameOrAfter(
+    dayjs(getLastPurchaseDate(item, item.lastPurchased)).add(
+      item.calculatedEstimate * 2,
+      'day',
+    ),
+  );
+
+  if (elapsedTime === true) {
+    return ['inactive', 'Inactive'];
+  } else if (estimatedCountdown <= 7) {
+    return ['soon', 'Within 7 days'];
+  } else if (estimatedCountdown <= 14) {
+    return ['kindOfSoon', 'Between 7 and 14 days'];
+  } else if (14 < estimatedCountdown) {
+    return ['notSoon', 'More than 14 days'];
+  } else if (isNaN(estimatedCountdown)) {
+    return ['notBought', 'Never Bought Yet'];
+  } else {
+    return null;
+  }
+};
+
+// Sort items by soonest to latest estimated repurchase
+const sortItems = (a, b) => {
+  if (a.calculatedEstimate === b.calculatedEstimate) {
+    return a.name.localeCompare(b.name, 'en', { sensitivity: 'base' });
+  } else if (
+    a.calculatedEstimate < b.calculatedEstimate ||
+    b.calculatedEstimate === undefined
+  ) {
+    return -1;
+  } else {
+    return 1;
+  }
+};
```

Comment:
I have a few thoughts on how to approach sorting inactive items to the bottom of the list. 

Before comparing the calculated estimate of both items, you would need to check if either or both items are considered inactive. 
- If both items are inactive, you compare their item names
- if only item A is inactive, it would come after item B (return 1)
- If only item B is inactive, it would come after item A (return -1)
- If neither items are inactive, you continue with comparing each calculated estimate

This will shift all of the inactive items to the bottom, even below items that have an undefined calculated estimate.

Also if I read correctly, the code on line 83 is returning a boolean if an item is inactive or not. You might be able to turn that into a helper function so that you can reuse it in your sorting function to check if either item is inactive. Hope I was able to explain my thought process! 😅

Reply:
We attempted to complete this, but were still not quite able to get there. We commented out the code that we tried to use, so maybe this is something we can all go over before merging (or the mentors can give more advice on prior to then). Thank you for the feedback though, this definitely helped us think through how we wanted to write this functionality!

Reply:
I should have clarified in my last comment, hope this makes things a little more clear. Tag me in Slack if you have any questions, I'll also be free to chat over zoom until 4PM PT today. 😀
```javascript
const sortItems = (a, b) => {
  // Sort by inactive item first
  const aInactive = dayjs().isSameOrAfter(
    dayjs(getLastPurchaseDate(a, a.lastPurchased)).add(
      a.calculatedEstimate * 2,
      'day',
    ),
  );

  const bInactive = dayjs().isSameOrAfter(
    dayjs(getLastPurchaseDate(b, b.lastPurchased)).add(
      b.calculatedEstimate * 2,
      'day',
    ),
  );

  if (aInactive && bInactive) {
    return a.name.localeCompare(b.name, { sensitivity: 'base' });
  } else if (aInactive) {
    return 1;
  } else if (bInactive) {
    return -1;
  }
  // Sort by calculatedEstimate
  // ...
```

Reply:
Terry, you did it again! This worked perfectly and looks great now! Thanks for showing how we could repurpose elaspedTime.

![MVP clap](https://media.giphy.com/media/UoRsCfqvGy1ELgKWwc/giphy.gif)